cluster/provision_gateway.sh

```diff
@@ -0,0 +1,675 @@
+         fi
+      fi
+      sed -i 's,# gateway=10.1.1.254,gateway='$defgw',g' $vrac
+      if isGceVM ; then
+          sed -i 's,# subnet_hosts_resolvable=0,subnet_hosts_resolvable=0,g' $vrac
+      fi
+      sed -i 's/# physical_interface=vnet0/physical_interface='$OPENCONTRAIL_VROUTER_INTF'/g' $vrac
+      sed -i 's/compute_node_address = 10.204.216.28/# compute_node_address = /g' $vrac
+  fi
+  wget -P /tmp https://raw.githubusercontent.com/Juniper/contrail-kubernetes/$ockver/cluster/contrail-vrouter-agent.manifest
+  vragentfile=/tmp/contrail-vrouter-agent.manifest
+  vrimg=$(cat $vragentfile | grep image | awk -F, '{print $1}' | awk '{print $2}')
+  echo $vrimg | xargs -n1 sudo docker pull
+  i=0
+  while true
+    do
+      dvrimg=$(docker images | grep -ow vrouter-agent)
+      if [ ! -z  "$dvrimg" ]; then
+         break
+      fi
+      sleep 5
+      ((i++))
+      if [ $i -eq 12 ]; then
+       if [ -d "/proc/${pullpid}" ]; then
+          pkill -TERM -P $pullpid
+          cnt=$(ps -ef|grep "docker pull" | grep vrouter-agent | wc -l)
+          if [ $cnt -gt 1 ]; then
+             log_info_msg "Restarting docker and retry pull vrouter image"
+             service docker restart
+          fi
+       fi
+       # give time for docker to initialize
+       sleep 60
+       log_info_msg "pulling of opencontrail/vrouter-agent image was not successful in the initial attempt."
+       (echo $vrimg | xargs -n1 sudo docker pull) & pullpid=$!
+       i=0
+      fi
+    done
+}
+
+function vr_agent_manifest_setup()
+{
+  # Keeping the manifest as in a minion
+  # there is no kubelet on a gateway node
+  if [ ! -f /etc/kubernetes/manifests ]; then
+     mkdir -p /etc/kubernetes/manifests
+  fi
+  vra_manifest="/etc/kubernetes/manifests/contrail-vrouter-agent.manifest"
+  if [ -f "$vra_manifest" ]; then
+     rm -f "$vra_manifest"
+  fi
+  # Wait for the control node to be up
+  # check 60 times in 5 min
+  cc=''
+  ifmapup=false
+  while true
+    do
+     cc=$(curl -s http://$OPENCONTRAIL_CONTROLLER_IP:8083/Snh_SandeshUVECacheReq?tname=NodeStatus | xmllint --format - | grep -ow "contrail-control")
+     if [ ! -z $cc ]; then
+         ifmapup=$(curl -s http://$OPENCONTRAIL_CONTROLLER_IP:8083/Snh_IFMapPeerServerInfoReq? | xmllint --format - | grep end_of_rib_computed | cut -d ">" -f2 | cut -d "<" -f1)
+     fi
+     if [ "$ifmapup" == true ]; then
+       log_info_msg "IFMAP and Contrail-Control are up are ready for processing agent subscriptions"
+       break
+     fi
+     sleep 3
+    done
+  mv /tmp/contrail-vrouter-agent.manifest /etc/kubernetes/manifests
+  
+  # start docker 
+  img=$(docker images | grep vrouter | awk '{print $1}')
+  ver=$(docker images | grep vrouter | awk '{print $2}')
+  
+  docker run --privileged -d -P --name contrail-vrouter-agent --net="host" -t -i -e sysimage=/host -v /etc/contrail:/etc/contrail -v /var/log/contrail:/var/log/contrail $img:$ver /usr/bin/contrail-vrouter-agent
```

Comment:
Lets wrap this with systemd / upstart. The kubenertes/contrib ansible code has configuration files for both.